src/Diagnostics.DataProviders/KustoSDKClient.cs

```diff
@@ -95,7 +95,23 @@ public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, str
             {
                 timeTakenStopWatch.Start();
                 var kustoClient = Client(cluster, database);
-                var result = await kustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
+                var kustoTask = kustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
+                if (_config.QueryShadowingClusterMapping != null && _config.QueryShadowingClusterMapping.TryGetValue(cluster, out var shadowClusters))
+                {
+                    foreach (string shadowCluster in shadowClusters)
+                    {
+                        try
+                        {
+                            var shadowKustoClient = Client(shadowCluster, database);
+                            shadowKustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
+                        }
+                        catch (Exception)
+                        {
+                            // swallow this exception
+                        }
+                    }
```

Comment:
Something I realize is if a sourceCluster can have say two testClusters then we can potentially use 3x the amount of threads. Or if there is 1 testCluster then we are potentially doubling the amount of threads though 1 task does not always equate to 1 thread.

I am curious on the affects of increased threads on the service. Maybe keep a close watch  of the thread count when this is deployed.

https://applens.azurewebsites.net/subscriptions/0e7b3183-4428-4c1f-93cd-cb2a26616a8a/resourceGroups/RuntimeHost-Prod-RG/providers/Microsoft.Web/sites/diag-runtimehost-prod-bay-001/detectors/threadcount?startTime=2021-08-30T16:59&endTime=2021-08-31T16:47